### DISCUSSION TITLE
MVC Example: Info & Warning Rules Only On Edit

### DISCUSSION THREAD

**Discussion by GreggThelen** [MarimerLLC/csla#3261] [Questions]
In the most recent MVC Example the Info and Warning rules appear to only work on Edit.
That is a fantastic feature; however, I'd like to see these work on Create as well.
Is that possible?

**Comment by rockfordlhotka** [comment]
Is it possible that in the example that `BusinessRules.CheckRules` isn't being called in the objects' create/createchild data portal operation methods?

By default, the create/createchild operations don't call `CheckRules` for performance reasons (and because some folks _don't_ want broken rules appearing on create screens).

```c#
  [Create]
  private void Create()
  {
    BusinessRules.CheckRules();
  }
```

**Reply by GreggThelen** [reply]
The check rules is in place, which is why the information rules appear, which is good.
The error rules don't appear until you try to submit and an error condition is recognized, which is good.

However, I cannot get the warning rules on a create for some reason - they do appear on an edit - just not a create.

**Comment by danielmartind** [comment]
In the sample the markup to show the warnings isn't on the create page. 
Add the below as in the edit view and you should see the warnings/info
`<span class="text-danger">@Html.ErrorFor(model => model.Name)</span>
<span class="text-warning">@Html.WarningFor(model => model.Name)</span>
 <span class="text-info">@Html.InformationFor(model => model.Name)</span>`
